docs/content/cookbook/oss-deployment-guide.md

```diff
@@ -0,0 +1,625 @@
+``` 
+
+#tab-2
+
+### Add SSH Key to Github Secrets
+We'll need to add an SSH key to our Github repository so that we can upload the tarball and run commands on our target server. Generate an SSH and make sure it's added to your server. Then, add the private key to your Github repository as a secret. To do this, go to your repository's settings page, click on "Secrets", and then click on "New repository secret". Name the secret "SSH_KEY" and paste the contents of your private key into the value field.
+
+:::list{type="success"}
+- Generate an SSH key
+- Go to your project settings
+- Go to "Secrets and Variables" (left menu)
+- Go to "Actions"
+- Click on "New repository secret" and add the secret `SSH_KEY`
+:::
+See the screenshot below for reference:
+![systemctl status image](./img/add-secrets.png)
+
+Github will automatically encrypt the secret for you. You can read more about Github secrets [here](https://docs.github.com/en/actions/reference/encrypted-secrets).
+
+#tab-3
+
+## Deployment Script
+
+### Why use Github Action?
+We want to deploy our applications based on a preconfigured setup. This reduces the chance of human error and frees up our time to work on more important things.
+
+To do this, we'll use Github Actions to build our application and deploy it to our server. Github Actions is a CI/CD tool that allows us to automate our development workflow. We'll use it to build our application, create a tarball of the application, upload the tarball to our server, extract the tarball on our server, and restart our services. For more information on Github Actions, check out the [official documentation](https://docs.github.com/en/actions).
+
+### What does this script do?
+::list{type="success"}
+- Builds the application
+- Creates a tarball of the application
+- Uploads the tarball to the server
+- Extracts the tarball on the server
+- Creates a symlink to the new release
+- Restarts the services
+- Cleans up old releases
+::
+
+::warning
+Notice we use an SSH_KEY secret in this deployment script. We'll need to add this to our Github repository. Check out the previous section for instructions on how to do this.
+::
+
+Let's go through the script step by step. At the end of this section, we'll provide the full script.
+
+### When to run the script
+We need to tell Github when to run our script. The `on` property in the script tells Github when to run the script. In this case, we want to run the script whenever we push to the main branch.
+```yaml
+on:
+  push:
+    branches:
+      - main
+```
+You can change this to only run when, for example, you create a realease tag. Check out the [Github Actions documentation](https://docs.github.com/en/actions) for more information.
+
+### Create deployment artifacts
+Now that we've told Github when to run our script, we need to tell it what to do. The `jobs` property in the script tells Github what to do. In this case, we want to create deployment artifacts.
+
+```yaml
+jobs:
```

Comment:
I like the step-by-step explanations, but for the code snippet it would be cleaner to just have a single code block (for readability and for copy/paste) imo.

Reply:
There is a full block at the end. I first attempted to just have explanations after the big block. But there's so much text it's easy to get lost imo. Do you think it'd be better to have that first? I tried that, but it felt more readable to first understand each bit then see it as a whole. I'm not married to it though. Does that block at the end satisfy what you're looking for?